dex/elrond_dex_farm/src/farm_token_merge.rs

```diff
@@ -0,0 +1,180 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use common_structs::{FarmTokenAttributes, GenericEsdtAmountPair};
+use farm_token::FarmToken;
+use token_merge::ValueWeight;
+
+use super::config;
+
+use super::farm_token;
+
+#[elrond_wasm_derive::module]
+pub trait FarmTokenMergeModule:
+    nft_deposit::NftDepositModule
+    + token_send::TokenSendModule
+    + farm_token::FarmTokenModule
+    + token_supply::TokenSupplyModule
+    + config::ConfigModule
+    + token_merge::TokenMergeModule
+{
+    fn merge_and_send_tokens(
+        &self,
+        #[var_args] opt_accept_funds_func: OptionalArg<BoxedBytes>,
+    ) -> SCResult<GenericEsdtAmountPair<Self::BigUint>> {
+        let caller = self.blockchain().get_caller();
+        let attrs = self.get_merged_farm_token_attributes(&caller, Option::None)?;
+        let farm_token_id = self.farm_token_id().get();
+
+        self.burn_deposit_tokens(&caller);
+        self.nft_deposit(&caller).clear();
+
+        self.nft_create_tokens(&farm_token_id, &attrs.current_farm_amount, &attrs);
+        self.increase_nonce();
+
+        let new_amount = attrs.current_farm_amount;
+        let new_nonce = self.farm_token_nonce().get();
+        self.send_nft_tokens(
+            &farm_token_id,
+            new_nonce,
+            &new_amount,
+            &caller,
+            &opt_accept_funds_func,
+        );
+
+        Ok(GenericEsdtAmountPair {
+            token_id: farm_token_id,
+            token_nonce: new_nonce,
+            amount: new_amount,
+        })
+    }
+
+    fn get_merged_farm_token_attributes(
+        &self,
+        caller: &Address,
+        replic: Option<FarmToken<Self::BigUint>>,
+    ) -> SCResult<FarmTokenAttributes<Self::BigUint>> {
+        let deposit_len = self.nft_deposit(caller).len();
+        require!(deposit_len != 0 || replic.is_some(), "No tokens to merge");
+
+        let mut index = 1;
+        let mut tokens = Vec::new();
+        let farm_token_id = self.farm_token_id().get();
+
+        while index <= deposit_len {
+            let entry = self.nft_deposit(caller).get(index);
+            require!(entry.token_id == farm_token_id, "Not a farm token");
```

Comment:
Some of these entries may be empty after calling `withdraw_token_from_deposit`, would it be worth skipping over them?

Reply:
Fixed. No more empty entries

Reply:
Resolved